src/common/arrow/physical_arrow_batch_collector.cpp

```diff
@@ -19,13 +19,13 @@ SinkFinalizeType PhysicalArrowBatchCollector::Finalize(Pipeline &pipeline, Event
 	if (total_tuple_count == 0) {
 		// Create the result containing a single empty result conversion
 		gstate.result = make_uniq<ArrowQueryResult>(statement_type, properties, names, types,
-		                                            context.GetClientProperties(), record_batch_size);
+		                                            context.GetClientProperties(), record_batch_size, context);
```

Comment:
We shouldn't need to keep a client context in the ArrowQueryResult at all - can we remove that?

Reply:
Maybe it makes sense to add a `weak_ptr<ClientContext>` to the client properties?